tap_salesforce/client.py

```diff
@@ -188,25 +179,88 @@ def get_records(
             limit_stm = f"LIMIT {limit}"
         else:
             limit_stm = ""
+        query = f"{select_stm} {from_stm} {where_stm} {order_by_stm} {limit_stm}"
+        return query
+
+    def field_chunker(
+        self, fields: List[str], size: int
+    ) -> Generator[List[str], None, None]:
+        field_chunk = []
+        length = 0
+        index = 0
+        for field in fields:
+            index += 1
+            length += len(field)
+            field_chunk.append(field)
+            if (length > size) or (index == len(fields)):
+                yield field_chunk
+                field_chunk = []
+                length = 0
+
+    def merge_records(
+        self, paginators: List[Generator[Dict, None, None]], table: Table
+    ) -> Generator[Dict, None, None]:
+        for records in zip(*paginators):
+            merged_record = {}
+            primary_key = None
+            for record in records:
+                if not primary_key:
+                    primary_key = record[table.primary_key]
+                if primary_key != record[table.primary_key]:
+                    raise PrimaryKeyNotMatch(
+                        f"couldn't merge records with different primary keys: {primary_key} and {record[table.primary_key]}"
+                    )
+                merged_record.update(record)
```

Comment:
```suggestion
            if not all(records[0][table.primary_key] == record[table.primary_key] for record in records):
                raise PrimaryKeyNotMatch("couldn't merge records with different primary keys")

            yield ChainMap(*records)
```

Reply:
I have never used chainmap and all function, it is very cool. Learned something new.